tests/CMakeLists.txt

```diff
@@ -37,10 +37,11 @@ set(TEST_TARGETS "")
 macro(add_test TEST_NAME)
     add_executable(
         ${TEST_NAME}
-        ${TEST_NAME}.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/bin/${TEST_NAME}.cpp
```

Comment:
The folder pair I'm used to seeing is `include` and `src`.

`bin` is typically reserved for binary files and other build artifacts. That being said, I normally feel comfortable blowing away `bin` folders, because I assume the build system will restore it.

Reply:
Works for me.